src/amp-story-player/amp-story-player-impl.js

```diff
@@ -800,24 +814,29 @@ export class AmpStoryPlayer {
 
   /**
    * Updates an iframe to the `inactive` state.
-   * @param {number} iframeIdx
+   * @param {!Element} story
    * @param {!IframePosition} position
    * @private
    */
-  updatePreviousIframe_(iframeIdx, position) {
+  updatePreviousIframe_(story, position) {
+    const iframeIdx = story[IFRAME_IDX];
     this.updateVisibilityState_(iframeIdx, VisibilityState.INACTIVE);
     this.updateIframePosition_(iframeIdx, position);
   }
 
   /**
    * Updates an iframe to the `current` state.
-   * @param {number} iframeIdx
+   * @param {!Element} story
    * @private
    */
-  updateCurrentIframe_(iframeIdx) {
-    this.updateVisibilityState_(iframeIdx, VisibilityState.VISIBLE);
-    this.updateIframePosition_(iframeIdx, IframePosition.CURRENT);
-    tryFocus(this.iframes_[iframeIdx]);
+  updateCurrentIframe_(story) {
+    const iframeIdx = story[IFRAME_IDX];
+    const iframeEl = this.iframes_[iframeIdx];
+    this.layoutIframe_(story, iframeEl, VisibilityState.VISIBLE).then(() => {
```

Comment:
This `layoutIframe_` is a preventive operation in case the user swiped to navigate before this new current story started its `layoutIframe_` (because it was waiting for the first story to load).